pkg/apis/controller/suggestions/v1alpha3/suggestion_types.go

```diff
@@ -0,0 +1,119 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	common "github.com/kubeflow/katib/pkg/apis/controller/common/v1alpha3"
+)
+
+// SuggestionSpec defines the desired state of Suggestion
+type SuggestionSpec struct {
+	// Number of suggestions requested
+	Suggestions *int32 `json:"suggestions,omitempty"`
+
+	//Algorithm settings set by the user in the experiment config
+	AlgorithmSpec *common.AlgorithmSpec `json:"algorithmSpec,omitempty"`
+}
+
+// SuggestionStatus defines the observed state of Suggestion
+type SuggestionStatus struct {
+	// Suggestion results
+	Assignments []TrialAssignment `json:"assignments,omitempty"`
+
+	// Represents time when the Suggestion was acknowledged by the Suggestion controller.
+	// It is not guaranteed to be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+
+	// Represents time when the Suggestion was completed. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// Represents last time when the Suggestion was reconciled. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	LastReconcileTime *metav1.Time `json:"lastReconcileTime,omitempty"`
+
+	// List of observed runtime conditions for this Suggestion.
+	Conditions []SuggestionCondition `json:"conditions,omitempty"`
+}
+
+type TrialAssignment struct {
+	// Suggestion results
+	Assignments []common.ParameterAssignment `json:"assignments,omitempty"`
+}
+
+// SuggestionCondition describes the state of the Suggestion at a certain point.
+// +k8s:deepcopy-gen=true
+type SuggestionCondition struct {
+	// Type of Suggestion condition.
+	Type SuggestionConditionType `json:"type"`
+
+	// Status of the condition, one of True, False, Unknown.
+	Status v1.ConditionStatus `json:"status"`
+
+	// The reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+
+	// A human readable message indicating details about the transition.
+	Message string `json:"message,omitempty"`
+
+	// The last time this condition was updated.
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty"`
+
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+}
+
+// SuggestionConditionType defines the state of a Suggestion.
+type SuggestionConditionType string
+
+const (
+	SuggestionCreated SuggestionConditionType = "Created"
+	SuggestionRunning SuggestionConditionType = "Running"
```

Comment:
Is Suggestion CR status always in Running state in our case?

Reply:
Currently, I think so. If we have other conditions, we can add them later.

Reply:
Sounds good.